crates/runtime/src/continuation.rs

```diff
@@ -382,6 +386,9 @@ pub mod baseline {
     /// Continues a given continuation.
     #[inline(always)]
     pub fn resume(instance: &mut Instance, contref: &mut VMContRef) -> Result<u32, TrapReason> {
+        // Trigger fuse
+        HAS_EVER_RUN_CONTINUATION.set(true);
```

Comment:
The following may be faster if `Cell::set` unconditionally writes to the internal memory cell, though, I don't know the specifics of the internals of `Cell::set`
```suggestion
   if HAS_EVER_RUN_CONTINUATION.get() {
        HAS_EVER_RUN_CONTINUATION.set(true);
   }
```

Reply:
Thinking about it again, maybe you should use `OnceCell`.

Reply:
Thinking yet again, there is a potential problem here (and regression). Consider the scenario: You run a continuation to completion (e.g. the unit continuation), and afterwards on the main stack you perform a wasm-to-host call that reenters wasm, but it would be prohibited from doing so because `HAS_EVER_RUN_CONTINUATION` is true even though we are not currently running on a continuation... or?

Reply:
> The following may be faster if Cell::set unconditionally writes to the internal memory cell, though, I don't know the specifics of the internals of Cell::set

Writing to the `Cell` here compiles down to a simple move instruction. So there isn't much of a point in making it conditional.

> Thinking yet again, there is a potential problem here (and regression). Consider the scenario: You run a continuation to completion (e.g. the unit continuation), and afterwards on the main stack you perform a wasm-to-host call that reenters wasm, but it would be prohibited from doing so because `HAS_EVER_RUN_CONTINUATION` is true even though we are not currently running on a continuation... or?

Please note that the `HAS_EVER_RUN_CONTINUATION` is not used by the mechanism to disallow certain forms of re-entering wasm. Instead, it is only used in the logic for generating backtraces: If we are using the baseline implementation and have ever run a continuation, we return an empty backtrace. In particular, the program you describe would still run normally. It's only the case that in your example returning to the main stack, we would still produce empty backtraces in the case of a trap. But I think that's acceptable; I'd rather not make the logic regarding this more complex.

Introducing the `HAS_EVER_RUN_CONTINUATION` flag is mostly just so that the full test suite passes when compiling with the baseline flag: There are plenty of non-wasmfx test cases that do something and inspect the resulting backtrace. These tests would fail if we unconditionally produced empty backtraces whenever the baseline feature is enabled.